src/main/scala/org/apache/sysml/api/dl/CaffeLayer.scala

```diff
@@ -190,8 +321,9 @@ class Scale(val param:LayerParameter, val id:Int, val net:CaffeNetwork) extends
   if(!param.getScaleParam.getBiasTerm) throw new LanguageException("Add \"scale_param { bias_term: true }\" to the layer " + param.getName)
   override def sourceFileName = null
   override def init(dmlScript: StringBuilder): Unit = {}
+  // TODO: Generalize this !!
   def forward(dmlScript: StringBuilder, isPrediction: Boolean): Unit = assign(dmlScript, out, X)
-  override def backward(dmlScript: StringBuilder, outSuffix:String): Unit = assign(dmlScript, dX + outSuffix, dout)
+  override def backward(dmlScript: StringBuilder, outSuffix:String): Unit = assignDoutToDX(dmlScript, outSuffix)
```

Comment:
To map this layer to [scale_shift2d.dml](https://github.com/apache/incubator-systemml/blob/master/scripts/nn/layers/scale_shift2d.dml), we will have to remove following code from batch_norm2d.dml:
https://github.com/apache/incubator-systemml/blob/master/scripts/nn/layers/batch_norm2d.dml#L118-L119